pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -551,3 +553,34 @@ func (exp *flowExporter) sendDataSet() (int, error) {
 	klog.V(4).Infof("Data set sent successfully. Bytes sent: %d", sentBytes)
 	return sentBytes, nil
 }
+
+func (exp *flowExporter) findFlowType(record flowexporter.FlowRecord) uint8 {
+	if record.Conn.Mark == openflow.ServiceCTMark || exp.ipInPodSubnets(record.Conn.TupleOrig.SourceAddress) && exp.ipInPodSubnets(record.Conn.TupleOrig.DestinationAddress) {
+		if record.Conn.SourcePodName == "" || record.Conn.DestinationPodName == "" {
+			return ipfixregistry.InterNode
+		}
+		return ipfixregistry.IntraNode
+	} else {
+		// Update Pod subnets to distinguish Pod-To-External flows.
+		if exp.nodeRouteController != nil {
+			exp.podSubnets = exp.nodeRouteController.GetPodSubnetsFromAllNodes()
+			klog.V(4).Infof("Updated Pod subnets: %v", exp.podSubnets)
+		}
+		if !exp.ipInPodSubnets(record.Conn.TupleOrig.SourceAddress) {
+			return ipfixregistry.FromExternal
+		} else if !exp.ipInPodSubnets(record.Conn.TupleOrig.DestinationAddress) {
+			return ipfixregistry.ToExternal
+		} else {
+			return exp.findFlowType(record)
+		}
+	}
+}
+
+func (exp *flowExporter) ipInPodSubnets(ip net.IP) bool {
+	for _, subnet := range exp.podSubnets {
+		if subnet.Contains(ip) {
+			return true
+		}
+	}
```

Comment:
Currently, we have fixed mask(same size) for all podCIDRs. Based on that we could optimize further.
Ok to keep this for now and consider the optimization later.